plugins/modules/sorcery.py

```diff
@@ -638,14 +728,33 @@ def main():
     elif params['state'] in ('absent', 'dispelled'):
         params['state'] = 'absent'
 
+    changed = {
+        'sorcery': (False, NA),
+        'grimoires': (False, NA),
+        'codex': (False, NA),
+        'spells': (False, NA)
+    }
+
     if params['update']:
-        update_sorcery(module)
+        changed['sorcery'] = update_sorcery(module)
+
+    if params['name'] and params['repository']:
+        changed['grimoires'] = manage_grimoires(module)
 
-    if params['update_cache'] or params['state'] == 'latest':
-        update_codex(module)
+    if params['update_cache']:
+        changed['codex'] = update_codex(module)
+
+    if params['name'] and not params['repository']:
+        changed['spells'] = manage_spells(module)
+
+    if list(filter(lambda x: changed.get(x)[0], changed)):
```

Comment:
I think you can also write this as follows:
```suggestion
    if any(x[0] for x in changed.values()):
```